InnerEye/Scripts/prepare_fastmri.py

```diff
@@ -0,0 +1,382 @@
+
+    # The data factory name. It must be globally unique.
+    data_factory_name = "fastmri-copy-data"
+
+    # Get either the Service Principal authentication, if those are set already, or use interactive auth in the browser
+    azureid_auth = get_azure_auth(azure_config)
+
+    # Create a data factory
+    adf_client = DataFactoryManagementClient(azureid_auth, azure_config.subscription_id)
+    df_resource = Factory(location=location)
+    print(f"Creating data factory {data_factory_name}")
+    df = adf_client.factories.create_or_update(azure_config.resource_group, data_factory_name, df_resource)
+    while df.provisioning_state != 'Succeeded':
+        df = adf_client.factories.get(azure_config.resource_group, data_factory_name)
+        time.sleep(1)
+    print("Data factory created")
+
+    # Create a linked service pointing to where the downloads come from
+    if is_unittest:
+        http_service = LinkedServiceResource(
+            properties=HttpLinkedService(url="https://github.com",
+                                         enable_server_certificate_validation=True,
+                                         authentication_type="Anonymous"))
+    else:
+        http_service = LinkedServiceResource(
+            properties=HttpLinkedService(url="https://fastmri-dataset.s3.amazonaws.com/",
+                                         enable_server_certificate_validation=True,
+                                         authentication_type="Anonymous"))
+    http_name = "AwsHttp"
+    adf_client.linked_services.create_or_update(resource_group_name=azure_config.resource_group,
+                                                factory_name=data_factory_name,
+                                                linked_service_name=http_name,
+                                                linked_service=http_service)
+    # Create a linked service that represents the sink (Azure blob storage)
+    blob_storage_name = "AzureBlob"
+    blob_storage = AzureBlobStorageLinkedService(connection_string=SecureString(value=connection_string))
+    blob_storage_service = LinkedServiceResource(properties=blob_storage)
+    adf_client.linked_services.create_or_update(resource_group_name=azure_config.resource_group,
+                                                factory_name=data_factory_name,
+                                                linked_service_name=blob_storage_name,
+                                                linked_service=blob_storage_service)
+
+    linked_blob_storage = LinkedServiceReference(reference_name=blob_storage_name)
+    linked_http = LinkedServiceReference(reference_name=http_name)
+
+    def download_and_uncompress(source_file_or_tuple: Union[str, Tuple[str, str]], target_folder: str) -> List[str]:
+        """
+        Downloads a file from AWS and stores them in blob storage in its compressed form.
+        Also, downloads the same file from AWS, uncompresses it, and writes the results to the given folder
```

Comment:
We download twice from AWS?

Reply:
Good spot. I did that initially, but that was changed. Updated comment.